test/distributed/org/apache/cassandra/distributed/test/accord/AccordTestBase.java

```
@@ -0,0 +1,166 @@
+
+        // Evict commands from the cache immediately to expose problems loading from disk.
+        sharedCluster.forEach(node -> node.runOnInstance(() -> AccordService.instance().setCacheSize(0)));
+
+        fn.accept(sharedCluster);
+
+        // Reset any messaging filters.
+        sharedCluster.filters().reset();
+    }
+
+    protected void test(FailingConsumer<Cluster> fn) throws Exception
+    {
+        test("CREATE TABLE " + currentTable + " (k int, c int, v int, primary key (k, c))", fn);
+    }
+
+    private static Cluster createCluster() throws IOException
+    {
+        // need to up the timeout else tests get flaky
+        // disable vnode for now, but should enable before trunk
+        return init(Cluster.build(2)
+                           .withoutVNodes()
+                           .withConfig(c -> c.with(Feature.NETWORK).set("write_request_timeout", "10s").set("transaction_timeout", "15s"))
+                           .withInstanceInitializer(ByteBuddyHelper::install)
+                           .start());
+    }
+
+    // TODO: Retry on preemption may become unnecessary after the Unified Log is integrated.
+    protected static SimpleQueryResult assertRowEqualsWithPreemptedRetry(Cluster cluster, Object[] row, String check, Object... boundValues)
+    {
+        SimpleQueryResult result = executeWithRetry(cluster, check, boundValues);
+        Assertions.assertThat(result.toObjectArrays()).isEqualTo(row == null ? new Object[0] : new Object[] { row });
+        return result;
+    }
+
+    protected static SimpleQueryResult executeWithRetry(Cluster cluster, String check, Object... boundValues)
+    {
+        try
+        {
+            return cluster.coordinator(1).executeWithResult(check, ConsistencyLevel.ANY, boundValues);
+        }
+        catch (Throwable t)
+        {
+            if (Throwables.getRootCause(t).toString().contains(Preempted.class.getName()))
+                return executeWithRetry(cluster, check, boundValues);
```

should we add delays?  If so can use `org.apache.cassandra.utils.Retry#retryWithBackoffBlocking(int, java.util.function.Supplier<A>)`

---

Preemption seems like a pretty rare case in my runs, and even if it does happen, I kind of like the fact that we're not waiting (and thereby going to easy on the protocol, hiding failures by allowing an excessive amount of time to pass).

---

Also, at some point in the future, I think this pre-emption checking might become unnecessary anyway. (i.e. Unified Log might make it unnecessary.) CC @bdeggleston